fiat-github/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java

```diff
@@ -44,139 +46,122 @@
-        } else if (e.getResponse().getStatus() == 404) {
-          log.error("404 when getting teams");
-          return result;
-        } else if (e.getResponse().getStatus() == 401) {
-          log.error(String.format("Cannot get GitHub organization %s teams: Not authorized.",
-                                  gitHubProperties.getOrganization()),
-                    e);
-          return result;
-        }
       }
+      log.debug("Got " + teamsPage.size() + " teams back. hasMorePages: " + hasMorePages);
     } while (hasMorePages);
 
-    log.debug("Found " + teams.size() + " teams in org.");
-    teams.forEach(t -> {
-      StringBuilder sb = new StringBuilder(userName).append(" is member of team ").append(t.getName());
-      if (isMemberOfTeam(t, userName)) {
-        sb.append(": true");
-        result.add(toRole(t.getSlug()));
+    return teams;
+  }
+
+  private List<Team> getTeamsInOrgPaginated(int page) {
+    List<Team> teams = new ArrayList<>();
+    try {
+      log.debug("Requesting page " + page + " of teams.");
+      teams = gitHubClient.getOrgTeams(gitHubProperties.getOrganization(),
+                                       page,
+                                       gitHubProperties.paginationValue);
+    } catch (RetrofitError e) {
+      if (e.getResponse().getStatus() != 404) {
+        handleNon404s(e);
       } else {
-        sb.append(": false");
+        log.error("404 when getting teams", e);
       }
-      log.debug(sb.toString());
-    });
+    }
 
-    return result;
+    return teams;
   }
 
-
-  private boolean isMemberOfTeam(GitHubMaster.Team t, String userName) {
+  private boolean isMemberOfTeam(Team t, String username) {
     String ACTIVE = "active";
     try {
-      GitHubMaster.TeamMembership response = master.getGitHubClient()
-                                                   .isMemberOfTeam(t.getId(), userName);
+      TeamMembership response = gitHubClient.isMemberOfTeam(t.getId(), username);
       return (response.getState().equals(ACTIVE));
     } catch (RetrofitError e) {
-      if (e.getKind() == RetrofitError.Kind.NETWORK) {
-        log.error(String.format("Could not find the server %s", master.getBaseUrl()), e);
-      } else if (e.getResponse().getStatus() == 401) {
-        log.error(String.format("Cannot check if $userName is member of %s teams: Not authorized.",
-                                t.getName()),
-                  e);
+      if (e.getResponse().getStatus() != 404) {
+        handleNon404s(e);
       }
     }
     return false;
   }
 
-  @Override
-  public void afterPropertiesSet() throws Exception {
-    Assert.state(gitHubProperties.getOrganization ()!= null, "Supply an organization");
+  private void handleNon404s(RetrofitError e) {
+    String msg = "";
+    if (e.getKind() == RetrofitError.Kind.NETWORK) {
+      msg = String.format("Could not find the server %s", gitHubProperties.getBaseUrl());
+    } else if (e.getResponse().getStatus() == 401) {
```

Comment:
Maybe it would be possible to handle the 403 when going over the rate limit? https://developer.github.com/v3/#rate-limiting this could be helpful to display the Github error message for troubleshooting